src/views/datasources/details/DataSourceYamlPage.tsx

```diff
@@ -0,0 +1,31 @@
+import * as React from 'react';
+import { RouteComponentProps } from 'react-router';
+
+import { V1beta1DataSource } from '@kubevirt-ui/kubevirt-api/containerized-data-importer/models';
+import Loading from '@kubevirt-utils/components/Loading/Loading';
+import { ResourceYAMLEditor } from '@openshift-console/dynamic-plugin-sdk';
+import { Bullseye } from '@patternfly/react-core';
+
+type DataSourceYAMLPageProps = RouteComponentProps<{
+  ns: string;
+  name: string;
+}> & {
+  obj?: V1beta1DataSource;
+};
+
+const DataSourceYAMLPage: React.FC<DataSourceYAMLPageProps> = ({ obj: dataSource }) => {
+  const loading = (
+    <Bullseye>
+      <Loading />
+    </Bullseye>
+  );
+  return !dataSource ? (
+    loading
+  ) : (
```

Comment:
maybe const loading is redundent? simply put content ?

Reply:
we should avoid popcorn loading, which is the current behavior in the vm details page, where data pops up everywhere in random timing.

Reply:
Wont we get the same? with or without const? i don't see a logic difference..

Reply:
I think we had some crashes so this fixed it, right @avivtur ?

Reply:
its was about React.suspense , not related to const loading :)